src/averageAges.js

```diff
@@ -55,7 +73,17 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = people.filter(child => onlyWithSon
+    ? child.sex === 'm' && people.some(mother => mother.name === child.mother)
+    : people.some(mother => mother.name === child.mother));
+
+  const averageAgeDiff = children.reduce((sum, child) => {
+    const mother = people.find(woman => woman.name === child.mother);
+
+    return sum + ((child.born - mother.born));
+  }, 0) / children.length;
+
+  return Math.round(averageAgeDiff * 100) / 100;
```

Comment:
You can use the 'toFixed' method to round the result to two decimal places. Example: return Number(averageAgeDiff.toFixed(2));

Reply:
thanks, good recomendation!